File: situational-awareness/frontend/src/types/mobile.ts
import { RiskFinding } from "@/types/risk";
import { TaskRun } from "@/types/task";

export type MobileDiscoveryEntry = {
  enabled: boolean;
  pending_jobs: number;
  running_jobs: number;
};

export type MobileOverview = {
  asset_total: number;
  online_assets: number;
  high_risk_findings: number;
  active_tasks: number;
  recent_tasks: TaskRun[];
  recent_risks: RiskFinding[];
  discovery_entry: MobileDiscoveryEntry;
};

File: situational-awareness/backend/app/schemas/mobile.py
from pydantic import BaseModel, Field

from app.schemas.risk import RiskFindingMobileRead
from app.schemas.task import TaskRunRead


class MobileDiscoveryEntryRead(BaseModel):
    enabled: bool = True
    pending_jobs: int = 0
    running_jobs: int = 0


class MobileOverviewRead(BaseModel):
    asset_total: int = 0
    online_assets: int = 0
    high_risk_findings: int = 0
    active_tasks: int = 0
    recent_tasks: list[TaskRunRead] = Field(default_factory=list)
    recent_risks: list[RiskFindingMobileRead] = Field(default_factory=list)
    discovery_entry: MobileDiscoveryEntryRead = Field(default_factory=MobileDiscoveryEntryRead)
